server/src/test/java/org/apache/druid/client/indexing/ClientCompactionRunnerInfoTest.java

```diff
@@ -148,6 +148,49 @@ public void testMSQEngineWithRollupFalseWithMetricsSpecIsInvalid()
     );
   }
 
+  @Test
+  public void testMSQEngineWithRollupTrueWithoutMetricsSpecIsInvalid()
+  {
+    DataSourceCompactionConfig compactionConfig = createCompactionConfig(
+        new DynamicPartitionsSpec(3, null),
+        Collections.emptyMap(),
+        new UserCompactionTaskGranularityConfig(null, null, true),
+        null
+    );
+    CompactionConfigValidationResult validationResult = ClientCompactionRunnerInfo.validateCompactionConfig(
+        compactionConfig,
+        CompactionEngine.NATIVE
+    );
+    Assert.assertFalse(validationResult.isValid());
+    Assert.assertEquals(
+        "MSQ: 'granularitySpec.rollup' must be false if 'metricsSpec' is null",
+        validationResult.getReason()
+    );
+  }
+
+  @Test
+  public void testMSQEngineWithUnsupportedMetricsSpecIsInValid()
+  {
+    // Aggregators having combiningFactory different from the aggregatorFactory are unsupported.
+    final String inputColName = "added";
+    final String outputColName = "sum_added";
+    DataSourceCompactionConfig compactionConfig = createCompactionConfig(
```

Comment:
Please rename the method `createCompactionConfig` to `createMSQCompactionConfig` to avoid confusion.

Reply:
Done